### ISSUE TITLE
Support loading multiple models at the same time

### ISSUE THREAD

**Issue by Picaso2** [ollama/ollama#2109]
is it possible to create one model from multiple models? or even load multiple models?

**Comment by Dbone29** [comment]
You can merge 2 models with another tool. https://huggingface.co/Undi95 He doe it with some models. After that you can a gguf file auf this model und use it in ollama when you want to. Ollama on its own isn't able to combine 2 modells.

**Comment by cmndcntrlcyber** [comment]
> You can merge 2 models with another tool. https://huggingface.co/Undi95 He doe it with some models. After that you can a gguf file auf this model und use it in ollama when you want to. Ollama on its own isn't able to combine 2 modells.

Do you happen to have a link or name of the tool?

**Comment by Dbone29** [comment]
There are many tools for this task, but unfortunately, I am not familiar enough to say which one is the best or what the differences between them are. However, here's an example of a tool that I came across last year:

https://github.com/arcee-ai/mergekit

**Comment by pdevine** [comment]
@Picaso2 other than the multimodal models we don't _yet_ support loading multiple models into memory simultaneously. What is the use case you're trying to do?

**Comment by mofanke** [comment]
> @Picaso2 other than the multimodal models we don't _yet_ support loading multiple models into memory simultaneously. What is the use case you're trying to do?

I encountered a similar requirement, and I want to implement a RAG (Retrieval-Augmented Generation) system. It requires using both an embedding model and a chat model separately. Currently, the implementation with Ollama requires constantly switching between models, which slows down the process. It would be much more efficient if there was a way to use them simultaneously.

**Comment by Picaso2** [comment]
Ultimately i would like to have an system that i can have a conversation with on various topics from science to politic to math.
________________________________

**Comment by Fruetel** [comment]
> @Picaso2 other than the multimodal models we don't _yet_ support loading multiple models into memory simultaneously. What is the use case you're trying to do?

I also have a use case for this. 

I'm using Crew.ai with Ollama. I have agents which need to use tools, such as search or document retrieval and then there are agents who work on data which are provided by the tool users. For tool using agent I use Hermes-2-Pro-Mistral, which is optimized for tool usage but not that smart with 7 billion parameters. Would be awesome to be able to load a smart Mixtral model for the thinking agents in parallel to the Hermes for the tool using ones.

**Comment by dizzyriver** [comment]
Same. I'd like llava for text to image and mixtral for reasoning

**Comment by alfi4000** [comment]
> Same. I'd like llava for text to image and mixtral for language reasoning

same

**Comment by alfi4000** [comment]
would It be possible to run several models 1 over gpu and the other ones over cpu and ram because I want to be able to run several models at once so if one of my family members are using Ollama over open webui and I do the same time it would be good that one runs on cpu and the other one one gpu!

**Comment by alfi4000** [comment]
> I have a rig with three graphics cards that I would like to run three separate models on simultaneously and have them group chat

Try run this: and edit the ip Adress 127.0.0.1 to your rigs ip Adress or just let it how it is and then add the ip Adress with Port to your instances for example I used open webui added those 3 and it managed one ip  Adress connection per Chat window so it should handle all 3 graphics car but only if you run it like I have you!: 

OLLAMA_HOST=127.0.0.1:11435 ollama serve & OLLAMA_HOST=127.0.0.1:11436 ollama serve & OLLAMA_HOST=127.0.0.1:11437 ollama serve

just an example you can use different ports but on one connection only one gpu and 1 llm not several other wise it will first finish first the second and then third!

**Comment by alfi4000** [comment]
> > I have a rig with three graphics cards that I would like to run three separate models on simultaneously and have them group chat
> 
> Try run this: and edit the ip Adress 127.0.0.1 to your rigs ip Adress or just let it how it is and then add the ip Adress with Port to your instances for example I used open webui added those 3 and it managed one ip Adress connection per Chat window so it should handle all 3 graphics car but only if you run it like I have you!:
> 
>
Linux (I tested in Ubuntu)
 OLLAMA_HOST=127.0.0.1:11435 ollama serve & OLLAMA_HOST=127.0.0.1:11436 ollama serve & OLLAMA_HOST=127.0.0.1:11437 ollama serve
> 
> just an example you can use different ports but on one connection only one gpu and 1 llm not several other wise it will first finish first the second and then third!

**Comment by leporel** [comment]
run in docker, stick containers separately with gpu1,gpu2 or cpu only, open-webui can work with multiply ollama instances

**Comment by dhiltgen** [comment]
> Can we have control over which model is run on which GPU?

This is something we can look at adding incrementally as this feature matures. Feel free to file a new issue and capture how you'd like it to work.

**Comment by dougy83** [comment]
> Can we have control over which model is run on which GPU?
You can create a new cpu-only model name using the following (e.g. for phi3 model)  `ollama show phi3 --model-file > phi3-cpuonly.modelfile`, editing that file to include `PARAMETER num_cpu 0`, and then using `ollama create -f phi3-cpuonly.modelfile phi3-cpuonly`
You then just reference the the phi3-cpuonly, and it loads into system RAM. You can call the file and model whatever you want.
